tests/gui/auth/sign_in/test_error_bar.py

```diff
@@ -1,8 +1,5 @@
-from PyQt5.QtWidgets import QApplication
-
 from securedrop_client.gui.auth.sign_in import LoginErrorBar
-
-app = QApplication([])
+from tests.helper import app  # noqa: F401
```

Comment:
This change makes sense to me based on the theory that we were actually grabbing an existing QApplication instance that was still exiting with `app = QApplication([])` rather than creating a new one - so when we tried to create QWidgets later the app no longer existed and hence the segfault.

Reply:
But I'm trying to understand why we need to manage the threads and add the wait still. Seems like this commit potentially fixes the entire issue, no? Are you thinking it's more of a best practices thing?

Reply:
They're two separate issues @creviera, the briefest answer is this build:
https://app.circleci.com/pipelines/github/freedomofpress/securedrop-client/2166/workflows/d6db676e-97f6-4bdb-81d9-8cfb45aa217c

If you look into why the Bullseye tests suite failed, you'll see that the **unit test suite succeeded** :green_apple: , but the **functional tests suite failed** :red_circle: ([error](https://app.circleci.com/pipelines/github/freedomofpress/securedrop-client/2166/workflows/d6db676e-97f6-4bdb-81d9-8cfb45aa217c/jobs/8556): `QThread: Destroyed while thread is still running`)

Reply:
In other terms:

Before #1515:

Buster unit tests: :green_apple: 
Buster functional tests: :green_apple: 
Bullseye unit tests: :red_circle: 
Bullseye functional tests: :red_circle: 
After #1515:

Buster unit tests: :green_apple: 
Buster functional tests: :green_apple: 
Bullseye unit tests: :green_apple: 
Bullseye functional tests: :red_circle: 

After this PR (#1512):

Buster unit tests: :green_apple: 
Buster functional tests: :green_apple: 
Bullseye unit tests: :green_apple: 
Bullseye functional tests: :green_apple:

Reply:
(I'll rebase the PR to "remove" from the diff the commit that is now in `main`.)